modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,321 @@
+        return peer.getMouseY();
+    }
+
+    /**
+     * Returns the current mouse (x, y) screen coordinates as a {@link Point2D}.
+     *
+     * @return the current mouse (x,y) screen coordinates
+     */
+    public Point2D getMousePosition() {
+        return new Point2D(getMouseX(), getMouseY());
+    }
+
+    /**
+     * Moves the mouse to the specified (x,y) screen coordinates.
+     *
+     * @param x screen coordinate x to move the mouse to
+     * @param y screen coordinate y to move the mouse to
+     */
+    public void mouseMove(double x, double y) {
+        peer.mouseMove(x, y);
+    }
+
+    /**
+     * Moves the mouse to the (x,y) screen coordinates specified by the
+     * given {@code location}.
+     *
+     * @param location the (x,y) coordinates to move the mouse to
+     */
+    public final void mouseMove(Point2D location) {
+        Objects.requireNonNull(location);
+        mouseMove(location.getX(), location.getY());
+    }
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     */
+    public void mousePress(MouseButton... buttons) {
+        peer.mousePress(buttons);
+    }
+
+    /**
+     * Releases the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to release
+     */
+    public void mouseRelease(MouseButton... buttons) {
+        peer.mouseRelease(buttons);
+    }
+
+    /**
+     * Clicks the specified {@link MouseButton}s.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #mousePress(MouseButton...)} followed by {@link #mouseRelease(MouseButton...)}.
+     * @param buttons the mouse buttons to click
+     */
+    public void mouseClick(MouseButton... buttons) {
+        mousePress(buttons);
+        mouseRelease(buttons);
+    }
+
+    /**
+     * Scrolls the mouse wheel by the specified amount. Positive {@code wheelAmt}
+     * scrolls up whereas negative scrolls down.
+     *
+     * @param wheelAmt the (signed) amount to scroll the wheel
+     */
+    public void mouseWheel(int wheelAmt) {
+        peer.mouseWheel(wheelAmt);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the specified screen coordinates of the
+     * primary screen.
+     *
+     * @param x the x coordinate to get the pixel color from
+     * @param y the y coordinate to get the pixel color from
+     * @return the pixel color at the specified screen coordinates
+     */
+    public Color getPixelColor(double x, double y) {
+        return peer.getPixelColor(x, y);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the screen coordinates of the primary screen
+     * specified by {@code location}.
+     *
+     * @param location the (x,y) coordinates to get the pixel color from
+     * @return the pixel color at the specified screen coordinates
+     */
+    public Color getPixelColor(Point2D location) {
+        Objects.requireNonNull(location);
```

Comment:
It might be good to add an `@throws NullPointerException` statement in the javadoc comments.  I realize we are not very consistent in doing this in other places.